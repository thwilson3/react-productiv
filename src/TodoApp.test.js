import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoApp from "./TodoApp";
import { initialTodos } from "./App";


describe("TodoApp", function () {
    it("renders without crashing", function () {
        // this is a low-value test, but better than nothing
        render(<TodoApp initialTodos={initialTodos} />);
    });

    it("renders todos", function () {
        const { container } = render(<TodoApp initialTodos={initialTodos} />);

        for (let todo of initialTodos) {
            expect(container).toContainHTML(todo.title);
            expect(container).toContainHTML(todo.description);
            expect(container).toContainHTML(todo.priority + "");
        }

    });

    it("renders placeholder text if no todos", function () {
        const { container } = render(<TodoApp />);

        expect(container).toContainHTML("You have no todos.");
        ;
    });

    it("can add a new todo", function () {
        const { container} = render(<TodoApp />);

        expect(container).toContainHTML("You have no todos.");

        const title = container.querySelector("#newTodo-title");
        const description = container.querySelector("#newTodo-description");
        const priority = container.querySelector("#newTodo-priority");
        const form = container.querySelector("#todo-form-new");

        fireEvent.change(title, { target: { value: "test title" }});
        fireEvent.change(description, { target: { value: "test description" }});
        fireEvent.change(priority, { target: { value: "2" }});
        fireEvent.submit(form);

        expect(container).toContainHTML("test title");
        expect(container).toContainHTML("test description");
        expect(container).toContainHTML("2");


    });

    it("can update a todo", function () {

    });

    it("can remove a todo", function () {

    });

    it("updates highest priority todo", function () {

    });


    it("matches snapshot", function () {
        const { container } = render(<TodoApp initialTodos={initialTodos} />);
        expect(container).toMatchSnapshot();
    });
});