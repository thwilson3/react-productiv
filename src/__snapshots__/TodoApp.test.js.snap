// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp matches snapshot 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                  id="edit-button-1"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
                id="todo-1"
              >
                <div>
                  <b>
                    Code!
                  </b>
                   
                  <small>
                    (priority: 
                    2
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Write some code
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                  id="edit-button-2"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
                id="todo-2"
              >
                <div>
                  <b>
                    Make dinner
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Cook something healthy
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                  id="edit-button-3"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
                id="todo-3"
              >
                <div>
                  <b>
                    Go to bed
                  </b>
                   
                  <small>
                    (priority: 
                    3
                    )
                  </small>
                </div>
                <div>
                  <small>
                    In bed by 11:15
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="mb-4"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
            id="todo-2"
          >
            <div>
              <b>
                Make dinner
              </b>
               
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div>
              <small>
                Cook something healthy
              </small>
            </div>
          </div>
        </section>
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
            id="todo-form-new"
          >
            <div
              class="mb-3"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
